pkg/skaffold/runner/build_deploy.go

```diff
@@ -127,11 +120,20 @@ func (r *SkaffoldRunner) DeployAndLog(ctx context.Context, out io.Writer, artifa
 		}
 	}
 
-	<-ctx.Done()
+	if r.runCtx.Opts.Tail || r.runCtx.Opts.PortForward.Enabled {
```

Comment:
why do we need this if?

Reply:
Because, without logs or port forwarding, we don't wait